doc/notes/189_reporting_ant.adoc

```diff
@@ -0,0 +1,224 @@
+shown as different colours (e.g. red for alarm, yellow for failure and
+green for success) shall make the proportions visually clear.
+
+The units shall be total alarm/failure/success over a noted time period
+(e.g. "today" or "last 24 hours").
+
+===== 3.1.1.4 List of Alarmed Jobs
+
+A list of job types that experienced alarm condition shall be displayed.
+The names of the jobs shall be shown in a readable format.  A counter of
+the number of alarms (over some period of time) shall be shown.  The list
+shall be sorted in descending order starting with the job type receiving
+the greatest incidence of alarms.
+
+===== 3.1.1.5 List of Failed Jobs
+
+A list of job types that failed shall be displayed.  The display format
+shall follow the format described in <<3.1.1.4 List of Alarmed Jobs>>.
+
+===== 3.1.1.6 List of Successful Jobs
+
+A list of job types that succeeded shall be displayed.  The display format
+shall follow the format described in <<3.1.1.4 List of Alarmed Jobs>>.
+
+===== 3.1.1.7 List of Retried Jobs
+
+A list of job types that experienced retries shall be displayed.  The
+names of the jobs shall be shown in a readable format.  A counter of the
+number of retries (over some period of time) shall be shown.  The list
+shall be sorted in descending order starting with the job type receiving
+the greatest incidence of retries.
+
+===== 3.1.1.8 Proportion of Happy versus Unhappy Jobs
+
+A relative proportion of happy jobs and unhappy jobs shall be displayed.
+A pie chart with the two types shown as different colours (e.g. blue for
+happy, purple for unhappy) shall make the proportions visually clear.
+
+The units shall be total happy/unhappy job instances over a noted time
+period (e.g. "today" or "last 24 hours").
+
+
+==== 3.1.2 Protocol Verifier Self-Reporting
+
+The Protocol Verifier provides metrics reflecting the behaviour of the
+monitored system(s).  However, it is useful to track a few key
+measurements supplying the current status of the Protocol Verifier itself.
+
+Following is a summary enumeration with details following.
+
+* PV uptime
+* number of running worker PVs
+* count of assigned jobs
+* count of unassigned jobs
+
+===== 3.1.2.1 Protocol Verifier Uptime
+
+The amount of time elapsed since the Protocol Verifier was started shall
+be displayed.
+
+The units shall be in time duration format.
+
+===== 3.1.2.2 Protocol Verifier Worker Count
+
+The Protocol Verifier scales up and down to handle various audit event
+stream loads efficiently.  An integer number representing the current
+number of Protocol Verifier "worker" instances shall be shown.  A max
+count and a minimum count shall display the maximum and minimum number
+respectively over a period of time (such as 24 hours, or since the
+Protocol Verifier started).
+
+There are no units to this measurement.
+
+===== 3.1.2.3 Count of Assigned Jobs (Jobs in Progress)
+
+The number of jobs which Job Management has assigned to a worker shall be
+displayed.
```

Comment:
As well as displaying the current state of the number of assigned jobs, is there any merit in also showing the average number of assigned jobs over a specified period of time.

This comment also applies to the number of unassigned jobs - in fact it might be more significant for unassigned jobs because, if we move to dynamically standing-up & standing down workers in the face of different loading, then this will be an indicator of the efficiency of that process.

Reply:
Good idea.  But let us keep this simplicity for now.  We can derive that information from what we have spec'd.